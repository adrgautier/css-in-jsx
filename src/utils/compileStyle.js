import {compile, serialize, middleware, stringify} from 'stylis';

const compileStyle = (style) => serialize(compile(style), middleware([stringify]));

export default compileStyle;