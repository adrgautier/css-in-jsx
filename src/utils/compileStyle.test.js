import compileStyle from './compileStyle';

describe('compileStyle', () => {
    it('should match snapshot', () => {
        // GIVEN
        const scopedStyleMock = `
        .scope {
            .card {
                display: block;
                & .title {
                    color: white;
                }
                {
                    background-color: red;
                }
            }
        }
        `;

        // WHEN
        const compiledStyle = compileStyle(scopedStyleMock);

        // THEN
        expect(compiledStyle).toMatchSnapshot();

    });
})