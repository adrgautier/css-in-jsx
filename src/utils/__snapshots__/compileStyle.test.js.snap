// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compileStyle should match snapshot 1`] = `".scope .card{display:block;background-color:red;}.scope .card .title{color:white;}"`;
